mne/simulation/tests/test_source.py

```diff
@@ -357,23 +356,50 @@ def test_source_simulator(_get_fwd_labels):
         counter += 1
     assert counter == 1
 
+    # Check validity of setting duration and start/stop parameters.
     half_ss = SourceSimulator(src, tstep, duration=0.5)
     for i in range(3):
         half_ss.add_data(mylabels[i], wfs[i], events[i])
     half_stc = half_ss.get_stc()
     assert_array_almost_equal(stc.data[:, :3], half_stc.data)
 
-    ss = SourceSimulator(src)
+    part_stc = ss.get_stc(start_sample=1, stop_sample=4)
+    assert part_stc.shape == (24, 4)
+    assert part_stc.times[0] == tstep
 
+    # Check validity of other arguments.
+    with pytest.raises(ValueError, match='start_sample must be'):
+        ss.get_stc(2, 0)
+    ss = SourceSimulator(src)
     with pytest.raises(ValueError, match='No simulation parameters'):
         ss.get_stc()
-
     with pytest.raises(ValueError, match='label must be a Label'):
         ss.add_data(1, wfs, events)
-
     with pytest.raises(ValueError, match='Number of waveforms and events '
                        'should match'):
         ss.add_data(mylabels[0], wfs[:2], events)
+    with pytest.raises(ValueError, match='duration must be None or'):
+        ss = SourceSimulator(src, tstep, tstep / 2)
+
+    # Verify first_samp functionality.
+    ss = SourceSimulator(src, tstep)
+    offset = 50
+    for i in range(3):  # events are offset, but first_samp = 0
+        events[i][:, 0] += offset
```

Comment:
```suggestion
        events[i, 0] += offset
```

Reply:
I'm not sure the events[i, 0] is correct!  The events array was (by a previous contributor to test_source.py) created as a list of separate event arrays... I'm a bit out of the loop now - was this change already reverted?